app/controllers/dashboard_controller.rb

```diff
@@ -10,6 +10,45 @@ def index
     end
   end
 
+  def create
+    Dashboard::Manager.add_widget_to_user(User.current, params[:widget])
```

Comment:
checking for duplicates would be difficult as the user can have many widgets of the same type with different parameters